packages/medusa/src/services/product-variant.ts

```diff
@@ -542,6 +551,36 @@ class ProductVariantService extends BaseService {
     })
   }
 
+  /**
+   * @param {object} selector - the query object for find
+   * @param {FindConfig<ProductVariant>} config - query config object for variant retrieval
+   * @return {Promise} the result of the find operation
+   */
+  async listAndCount(
+    selector: FilterableProductVariantProps,
+    config: FindConfig<ProductVariant> = { relations: [], skip: 0, take: 20 }
+  ): Promise<[ProductVariant[], number]> {
+    const variantRepo = this.manager_.getCustomRepository(
+      this.productVariantRepository_
+    )
+
+    const { q, query, relations } = this.prepareListQuery_(selector, config)
+
+    if (q) {
+      const qb = this.getFreeTextQueryBuilder_(variantRepo, query, q)
+      const [raw, count] = await qb.getManyAndCount()
+
+      const products = await variantRepo.findWithRelations(
+        relations,
+        raw.map((i) => i.id),
+        query.withDeleted ?? false
+      )
+      return [products, count]
```

Comment:
To go along with the above suggestion:
```suggestion
      return [variants, count]
```